metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/MetricAnomalyDetector.java

```diff
@@ -1,20 +1,146 @@
 
-  @Override
-  protected void doStop() {}
+  static final String TENANT_ID_KEY = "x-tenant-id";
 
-  @Override
-  public boolean healthCheck() {
-    return false;
+  private final MetricQueryBuilder metricQueryBuilder;
+  private final QueryServiceClient queryServiceClient;
+  private final int qsRequestTimeout;
+
+  MetricAnomalyDetector(Config appConfig) {
+    AttributeServiceClientConfig asConfig = AttributeServiceClientConfig.from(appConfig);
+    ManagedChannel attributeServiceChannel =
+        ManagedChannelBuilder.forAddress(asConfig.getHost(), asConfig.getPort())
+            .usePlaintext()
+            .build();
+    AttributeServiceClient asClient = new AttributeServiceClient(attributeServiceChannel);
+
+    Config qsConfig = appConfig.getConfig(QUERY_SERVICE_CONFIG_KEY);
+    queryServiceClient = new QueryServiceClient(new QueryServiceConfig(qsConfig));
+    qsRequestTimeout =
+        appConfig.hasPath(REQUEST_TIMEOUT_CONFIG_KEY)
+            ? appConfig.getInt(REQUEST_TIMEOUT_CONFIG_KEY)
+            : DEFAULT_REQUEST_TIMEOUT_MILLIS;
+
+    metricQueryBuilder = new MetricQueryBuilder(asClient);
+  }
+
+  void process(AlertTask alertTask) {
+    MetricAnomalyEventCondition metricAnomalyEventCondition;
+    if (alertTask.getEventConditionType().equals(METRIC_ANOMALY_EVENT_CONDITION)) {
+      try {
+        metricAnomalyEventCondition =
+            MetricAnomalyEventCondition.parseFrom(alertTask.getEventConditionValue());
+      } catch (InvalidProtocolBufferException e) {
+        LOGGER.error("Exception while parsing event condition", e);
+        return;
+      }
+    } else {
+      LOGGER.debug("EventConditionType:{}", alertTask.getEventConditionType());
+      return;
+    }
+
+    QueryRequest queryRequest =
+        metricQueryBuilder.buildMetricQueryRequest(
+            metricAnomalyEventCondition.getMetricSelection(),
+            alertTask.getLastExecutionTime(),
+            alertTask.getCurrentExecutionTime(),
+            alertTask.getTenantId());
+
+    ViolationCondition violationCondition =
+        metricAnomalyEventCondition.getViolationConditionList().get(0);
+
+    Iterator<ResultSetChunk> iterator =
+        executeQuery(Map.of(TENANT_ID_KEY, alertTask.getTenantId()), queryRequest);
+
+    LOGGER.debug(
+        "Starting rule evaluation for rule Id {} start {} & end time {}",
+        alertTask.getEventConditionId(),
+        Instant.ofEpochMilli(alertTask.getLastExecutionTime()),
+        Instant.ofEpochMilli(alertTask.getCurrentExecutionTime()));
+
+    if (violationCondition.hasStaticThresholdCondition()) {
+      evaluateForStaticThreshold(iterator, violationCondition);
+    }
+  }
+
+  void evaluateForStaticThreshold(
+      Iterator<ResultSetChunk> iterator, ViolationCondition violationCondition) {
+    int dataCount = 0, violationCount = 0;
+    while (iterator.hasNext()) {
+      ResultSetChunk resultSetChunk = iterator.next();
+      for (Row row : resultSetChunk.getRowList()) {
```

Comment:
Don't we have to use `resultSetChunk`'s metadata - number of columns and at all.

Reply:
Using it now